php/helpers/helpers.php

```diff
@@ -65,6 +65,17 @@ function eq($a, $b) {
   return false;
 }
 
+/**
+ * @param $a
+ * @param $b
+ * @return float
+ */
+function zero_fill_right_shift($a, $b) {
+    if($a>=0) return (float)($a>>$b);
+    if($b==0) return (float)((($a>>1)&0x7fffffff)*2+(($a>>$b)&1));
+    return (float)(((~$a)>>$b)^(0x7fffffff>>($b-1)));
```

Comment:
Quick note: ;et's match the code style here
```suggestion
    if ($a >= 0) return (float)($a >> $b);
    if ($b == 0) return (float)((($a >> 1) & 0x7fffffff) * 2 + (($a >> $b) & 1));
    return (float)(((~$a) >> $b) ^ (0x7fffffff >> ($b - 1)));
```

As for the math here, to be honest I don't totally understand. If the tests are passing, I'm OK to ship this, but do you have a reference or a stack overflow post or anything that might have more details?

Reply:
About the formatting, I wonder if we should implement [Prettier for PHP](https://loilo.github.io/prettier-php-playground/) in this project. I'll check into it.

Reply:
Formatting isn't something we should have to think about, it should be automatic and seamless. I created issue #17 to add [Prettier](https://github.com/prettier/plugin-php) for this purpose.

Reply:
Oh sorry, didn't see your question. I used this [answer](https://stackoverflow.com/a/27263298) but have no clue about the mathematics behind it.